plugins/modules/identity/bitwarden.py

```diff
@@ -0,0 +1,371 @@
+                base['login'][key] = value
+                ret_val['changed'] = True
+
+        # Update the base notes section.
+        if do_updates and self._notes is not None:
+            if base['notes'] != self._notes:
+                base['notes'] = self._notes
+                ret_val['changed'] = True
+
+        if ret_val['changed']:
+            # Encode it using bw.
+            out, _stderr = Client.run(['encode'], stdin=json.dumps(base))
+            ret_val['ansible_module_results'] = Client.run_json(bw_args, stdin=out)
+        else:
+            # Return cached value, which should be identical, if no change occured.
+            ret_val['ansible_module_results'] = base
+
+        return ret_val
+
+    def _handle_present_folder(self):
+        ret_val = {'changed': False}
+
+        if self.folder.exists:
+            # A folder with this name already exists; return it.
+            ret_val['ansible_module_results'] = self.folder.dict
+
+        else:
+            # 'No Folder' is a reserved name in Bitwarden -- we shouldn't create a folder
+            # with that name.
+            if self.folder.name in (None, '', 'No Folder'):
+                raise AssertionError('Illegal folder name used')
+
+            # Fill in a template for a folder object to fill in.
+            template = Client.run_json(['get', 'template', 'folder'])
+            template['name'] = self.folder.name
+
+            # Encode it using bw.
+            out, _stderr = Client.run(['encode'], stdin=json.dumps(template))
+
+            # Create the folder.
+            bw_args = ['create', 'folder'] + self.organization.bw_filter_args
+            ret_val['ansible_module_results'] = Client.run_json(bw_args, stdin=out)
+            ret_val['changed'] = True
+
+        return ret_val
+
+    def _handle_present(self, update=True):
+        # Use a lookup table to determine handler for desired target.
+        dispatch = defaultdict(
+            _err('Unknown target'),
+            organization=_err('Creating organizations not supported'),
+            folder=self._handle_present_folder,
+            item=lambda: self._handle_present_item(update),
+        )
+
+        return dispatch[self._target]()
+
+    def _handle_created(self):
+        # Use a lookup table to determine handler for desired target.
+        dispatch = defaultdict(
+            _err('Unknown target'),
+            organization=_err('Creating organizations not supported'),
+            folder=self._handle_present_folder,
+            item=self._handle_created_item,
+        )
+
+        return dispatch[self._target]()
+
+    def _handle_absent(self):
+        ret_val = {'changed': False}
+
+        org_args = self.organization.bw_filter_args
+
+        if self._target == 'organization':
+            raise BitwardenException('Removing organizations not supported.')
+
+        if self._target in ('folder', 'item'):
```

Comment:
I don't _think_ the `required_if` is needed since `target` is always one of `'folder'` or `'item'` already; I previously included the check in case someone was incorrectly trying to call the Python code directly.